src/java/org/apache/cassandra/index/Index.java

```diff
@@ -765,6 +779,22 @@ default void invalidate() { }
          * @return the SSTable components created by this group
          */
         Set<Component> getComponents();
+
+        /**
+         * Validates all indexes in the group against the specified SSTables. 
+         * 
+         * @param sstables SSTables for which indexes in the group should be built
+         * @param throwOnIncomplete whether to throw an error if any index in the group is incomplete
```

Comment:
`or not valid` at the end?

Reply:
We throw on a validation error whether or not `throwOnIncomplete == true`